.pre-commit-config.yaml

```diff
@@ -0,0 +1,37 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
+    hooks:
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+    -   id: check-yaml
+    -   id: check-added-large-files
+
+  - repo: local
+    hooks:
+      - id: clang-format-fix
+        name: clang-format-fix
+        entry: clang-format-18
+        files: ^(src/|include/)
+        types_or: [c++, c]
+        language: system
+        args: ['-i']
+        stages: [manual]
+
+  - repo: https://github.com/pocc/pre-commit-hooks
```

Comment:
Project have not been updated been updated for 3 years. 

Seem not to be actively maintained, latest respond to issue tracker was in 2023.
This seem to just be a wrapper around calling clang-format, but an alternative maybe preferred.

Reply:
Moved clang-format to [`an actively maintained clang-format package`](https://github.com/pre-commit/mirrors-clang-format) in fc4d919a02a4f43b9e335ae623a80bb035c1f592

Reply:
No alternative for `clang-tidy`, but it have more problems to worry about, see below.